### PR TITLE
Add MPI build matrix

### LINKED ISSUE #43: Add mpi4py to build
There is a [switch](https://github.com/Unidata/netcdf4-python/blob/de23edcf957dee198e80c9e70a6469371fde445b/setup.py#L487-L491) in netcdf4's `setup.py` script that disables the parallel IO feature when `mpi4py` is not available. Would it work to add `mpi4py` to the build step in this feedstock? I'm personally not sure of the full implications of this so this is really just a question about feasibility.

### PR DESCRIPTION
**Pull request by xylar**
Also adds a parallel test.

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes #43

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
After rerendering, the feedstock is pulling in libnetcdf 4.7.2 rather than 4.7.1.  @conda-forge/netcdf4, any suggestions on how to keep this pinned correctly but still pull in the right MPI version?

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
Answering my own question from above:
https://github.com/conda-forge/h5py-feedstock/blob/master/recipe/meta.yaml#L49-L52
It looks like the solution to getting pinnings is to list each dependency twice.

**Comment by xylar** [comment]
@conda-forge/netcdf4, I think we're there.  Anyone want to do some final testing before I merge this?

File: recipe/parallel_test.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# reopen the file read-only again, check the data.
+# leave out the comm and info kwargs to check that the defaults
+# (MPI_COMM_WORLD and MPI_INFO_NULL) work.
+nc = Dataset('parallel_test.nc', parallel=True)
+if rank == 3:
+    assert 2 * rank == nc['var'][rank]
+else:
+    assert rank == nc['var'][rank]
+nc.close()
```
Comment (ocefpaf) [inline recipe/parallel_test.py:40]: Nice test!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -25,24 +46,37 @@ requirements:
...
+  files:
+    - parallel_test.py  # [mpi != 'nompi']
   imports:
     - netCDF4
     - cftime
   commands:
     - ncinfo -h
     - nc4tonc3 -h
     - nc3tonc4 -h
+    - mpirun -np 4 python parallel_test.py  # [mpi == 'mpich']
```
Comment (ocefpaf) [inline recipe/meta.yaml:78]: I Interesting the `4` just worked here. I thought you would need to use ``{{ CPU_COUNT }}`` to make it work properly on the CIs.

**Comment by ocefpaf** [comment]
@xylar this is great! Thanks so much for doing this. I can see all the PRs that lead you here and it was not easy!

**Comment by xylar** [comment]
@ocefpaf, thanks again!

**Pull request merged**